js/interface.js

```diff
@@ -307,32 +311,115 @@
     return Fliplet.API.request({
       url: [
         'v1/widgets?include_instances=true&tags=type:menu',
+        '&include_all_versions=true',
         '&appId=' + Fliplet.Env.get('appId'),
-        '&organizationId=' + (Fliplet.Env.get('organizationId') || '')
+        '&organizationId=' + (organizationId || '')
       ].join('')
     }).then(function(response) {
       return Promise.resolve(response.widgets);
     });
   }
 
+  /**
+   * Function to filter organizationMenus by version.
+   * @param {Array} data - array of organizationMenus
+   * @param {Object} latestVersionMenu - current active menu
+   * @returns {Object} latestVersionMenu - object which contains the latest version of current menu package type
+   */
+
+  function sortByVersion(data, latestVersionMenu) {
+    var previousVersion;
+
+    data.forEach(function(menu) {
+      // Compare each record with the previous one by version to find the newest version.
+      if (previousVersion && previousVersion.version !== menu.version) {
+        latestVersionMenu = (previousVersion.version > menu.version) ? previousVersion : menu;
+
+        return;
+      }
+
+      previousVersion = menu;
+
+      // Always write the system menu first, then if iteration finds the newest version, latestVersionMenu will be overwritten.
+      if (!latestVersionMenu) {
+        latestVersionMenu = menu;
+      }
+    });
+
+    return latestVersionMenu;
+  }
+
+  /**
+   * Function which filtering current array of menus by version and created a new one.
+   * @param {Array} menus - array of menus which comes from api
+   * @returns {Array} result - new filtered array by version
+   */
+
+  function generateMenuList(menus) {
+    var result = [];
+    var hasActiveMenu = false;
+
+    // Filtering current menus array by organizationId to create array of system menus.
+    var systemMenus = menus.filter(function(item) {
+      return item.organizationId !== organizationId;
+    });
+
+    // Cycle iterating the array of system menus to fulfill the necessary conditions.
+    systemMenus.forEach(function(item) {
+      var latestVersionMenu;
+
+      // Creating an array of all custom entries for one package of the system menu.
+      var organizationMenus = menus.filter(function(menu) {
```

Comment:
I don't understand this logic.

Organization menus, by definition, are menus that have an `organizationId` assigned to the widget. There doesn't need to be any loop within the system menus to compare with a system menu package name.

Reply:
@tonytlwu 
Here I create an array that contains the menu of the same type in order to further sort their versions

Reply:
Menus of the same type should have matching packages.

You're using `menu.package.includes(item.package)`, which suggests the following two menus will be considered the same type based on their packages, but they are not.

1. `com.fliplet.foo.bar`
2. `com.fliplet.foo`

Reply:
@tonytlwu, the logic was updated